specification.md

```diff
@@ -0,0 +1,209 @@
+The `Tracer` interface creates `Span`s and understands how to `Inject`
+(serialize) and `Extract` (deserialize) them across process boundaries.
+Formally, it has the following capabilities:
+
+#### Start a new `Span`
+
+Required parameters
+    - An **operation name**, a human-readable string which
+      concisely represents the work done by the Span (for example, an RPC
+      method name, a function name, or the name of a subtask or stage within a
+      larger computation). The operation name should be **the most general
+      string that identifies a (statistically) interesting class of `Span`
+      instances**. That is, `"get_user"` is better than `"get_user/314159"`.
+
+For example, here are potential **operation names** for a `Span` that gets hypothetical account information:
+
+| Operation Name | Guidance |
+|:---------------|:--------|
+| `get` | Too general |
+| `get_account/792` | Too specific |
+| `get_account` | Good, and `account_id=792` would make a nice **`Span` tag** |
+
+Optional parameters
+    - Zero or more **references**, including a shorthand for `CHILD_OF` and
+      `FOLLOWS_FROM` reference types if possible.
+    - An optional explicit **start timestamp**; if omitted, the current
+      walltime is used by default
+    - Zero or more **tags**
+
+**Returns** a `Span` instance that's already started (but not `Finish`ed)
+
+#### Inject a `SpanContext` into a carrier
+
+Required parameters
+    - A **`SpanContext`** instance
+    - A **format** descriptor (typically but not necessarily a string constant) which tells the `Tracer` implementation how to encode the `SpanContext` in the carrier parameter
+    - A **carrier**, whose type is dictated by the **format**. The `Tracer` implementation will encode the `SpanContext` in this carrier object according to the **format**.
+
+#### Extract a `SpanContext` from a carrier
+
+Required parameters
+    - A **format** descriptor (typically but not necessarily a string constant) which tells the `Tracer` implementation how to decode `SpanContext` from the carrier parameter
+    - A **carrier**, whose type is dictated by the **format**. The `Tracer` implementation will decode the `SpanContext` from this carrier object according to **format**.
+
+**Returns** a `SpanContext` instance suitable for use as a **reference** when starting a new `Span` via the `Tracer`.
+
+#### Note: required **format**s for injection and extraction
+
+Both injection and extraction rely on an extensible **format** parameter that dictates the type of the associated "carrier" as well as how a `SpanContext` is encoded in that carrier. All of the following **format**s must be supported by all Tracer implementations.
+    - **Text Map**: an arbitrary string-to-string map with an unrestricted character set for both keys and values
+    - **HTTP Headers**: a string-to-string map with keys and values that are suitable for use in HTTP headers. That is, keys must not require strict casing and values must be URL-encoded
```

Comment:
http header values are documented as ISO-8859-1/ASCII octets or [mime encoded](https://tools.ietf.org/html/rfc2047) values. Since we aren't doing that, the "URL-encoded" part should be a link explaining why senders and receivers be looking at that vs what's normal.